build.gradle.kts

```diff
@@ -37,6 +40,38 @@ tasks.withType<Test> {
 
 tasks.withType<JavaCompile>().configureEach {
     options.encoding = "UTF-8"
+    dependsOn.add(processPluginPropertiesFile)
+}
+
+val processPluginPropertiesFile = tasks.register<DefaultTask>("processPluginPropertiesFile") {
```

Comment:
I can see that when moving from Groovy to Kotlin for the Gradle configuration, the code updating the plugin.properties file automatically has been removed: https://github.com/CycloneDX/cyclonedx-gradle-plugin/pull/305/files#diff-49a96e7eea8a94af862798a45174e6ac43eb4f8b4bd40759b5da63ba31ec3ef7L38

It would be nice to have it automatically updated. An option is to add the task back using the Kotlin syntax and make it part of the compilation step, but it would cause continuous changes to the properties file even if nothing changed (due to the autogenerated timestamp) resulting in a poor development experience. I have drafted this alternative solution to change the file only if there was any actual change. 

This custom task could stay here or, if we want it defined in its own file, it could be moved to a `buildSrc` folder within the project.